src/auction/Auction.sol

```diff
@@ -58,15 +58,8 @@ contract Auction is IAuction, VersionedContract, UUPS, Ownable, ReentrancyGuard,
     /// @param _token The ERC-721 token address
     /// @param _founder The founder responsible for starting the first auction
     /// @param _treasury The treasury address where ETH will be sent
-    /// @param _duration The duration of each auction
-    /// @param _reservePrice The reserve price of each auction
-    function initialize(
-        address _token,
-        address _founder,
-        address _treasury,
-        uint256 _duration,
-        uint256 _reservePrice
-    ) external initializer {
+    /// @param _data The encoded auction settings
+    function initialize(address _token, address _founder, address _treasury, bytes calldata _data) external initializer {
```

Comment:
typically i prefer structs in this situation as noted in the diff notes due to sometimes lower code size and easier initialization on the front-end.

Reply:
The _data parameter is a struct so clients will be using a struct to encode it still. the address params come from the manager contract. would a struct still be useful for the code size aspect?